File: pocketr-java.api/src/main/java/com/decrux/pocketr/api/config/security/SecurityConstants.java
package com.decrux.pocketr.api.config.security;

public enum SecurityConstants {
    BACKEND_LOGIN_PROCESSING_URL("/v1/user/login"),
    BACKEND_LOGOUT_PROCESSING_URL("/v1/user/logout"),
    BACKEND_REGISTER_ENDPOINT("/v1/user/register"),
    GET_CSRF_TOKEN_ENDPOINT("/v1/internal/csrf-token"),
    FRONTEND_LOGIN_URL("/frontend/login"),
    FRONTEND_REGISTER_URL("/frontend/registration"),
    FRONTEND_STATIC_ASSETS("/frontend/assets/**"),
    FRONTEND_FAVICON("/frontend/favicon.ico"),
    FRONTEND_LANDING_PAGE("/frontend/index.html");

    private final String value;

    SecurityConstants(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}

File: pocketr-api/src/main/kotlin/com/decrux/pocketr/api/config/security/SecurityConstants.kt
package com.decrux.pocketr.api.config.security

enum class SecurityConstants(
    val value: String,
) {
    BACKEND_LOGIN_PROCESSING_URL("/api/v1/user/login"),
    BACKEND_LOGOUT_PROCESSING_URL("/api/v1/user/logout"),
    BACKEND_REGISTER_ENDPOINT("/api/v1/user/register"),
    GET_CSRF_TOKEN_ENDPOINT("/api/v1/internal/csrf-token"),
    BACKEND_API_ROOT("/api"),
    BACKEND_API("/api/**"),
    FRONTEND_LOGIN_URL("/login"),
    FRONTEND_STATIC_ASSETS("/assets/**"),
    FRONTEND_FAVICON("/favicon.ico"),
    FRONTEND_INDEX("/index.html"),
    FRONTEND_ROOT("/"),
}
